src/sagas/networkSettings.js

```diff
@@ -0,0 +1,186 @@
+import { all, takeEvery, put, call, race, take, delay } from 'redux-saga/effects';
+import { config } from '@hathor/wallet-lib';
+import { isEmpty } from 'lodash';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { featureToggleUpdate, networkSettingsUpdateFailure, networkSettingsUpdateSuccess, reloadWalletRequested, types } from '../actions';
+import { HTTP_REQUEST_TIMEOUT, NETWORK, networkSettingsKey, NETWORK_TESTNET, STAGE, STAGE_DEV_PRIVNET, STAGE_TESTNET } from '../constants';
+import { getFullnodeNetwork, getWalletServiceNetwork } from './helpers';
+import { STORE } from '../store';
+
+export function* updateNetworkSettings(action) {
+  const {
+    nodeUrl,
+    explorerUrl,
+    walletServiceUrl,
+  } = action.payload || {};
+  let walletServiceWsUrl = action.payload?.walletServiceWsUrl;
+
+  // validates input emptyness
+  if (isEmpty(action.payload)) {
+    yield put(networkSettingsUpdateFailure());
+    return;
+  }
+
+  // validates explorerUrl
+  // - required
+  // - should have a valid URL
+  if (isEmpty(explorerUrl) || invalidUrl(explorerUrl)) {
+    yield put(networkSettingsUpdateFailure());
+    return;
+  }
+
+  // validates nodeUrl
+  // - required
+  // - should have a valid URl
+  if (isEmpty(nodeUrl) || invalidUrl(nodeUrl)) {
+    yield put(networkSettingsUpdateFailure());
+    return;
+  }
+
+  // validates walletServiceUrl
+  // - optional
+  // - should have a valid URL, if given
+  if (walletServiceUrl && invalidUrl(walletServiceUrl)) {
+    yield put(networkSettingsUpdateFailure());
+    return;
+  }
+
+  // validates walletServiceWsUrl
+  // - optional
+  // - should have a valid URL, if given
+  if (walletServiceWsUrl && invalidUrl(walletServiceWsUrl)) {
+    yield put(networkSettingsUpdateFailure());
+    return;
+  }
+
+  // NOTE: Should we allow that all the URLs be equals?
+  // In practice they will never be equals.
+
+  const oldExplorerUrl = config.getExplorerServiceBaseUrl();
+  const oldNodeUrl = config.getServerUrl();
+  const oldWalletServiceUrl = config.getWalletServiceBaseUrl();
+  const oldWalletServiceWsUrl = config.getWalletServiceBaseWsUrl();
+
+  config.setExplorerServiceBaseUrl(explorerUrl);
+  config.setServerUrl(nodeUrl);
+
+  // - walletServiceUrl has precedence
+  // - nodeUrl as fallback
+  let network;
+  if (walletServiceUrl) {
+    const _url = new URL(walletServiceUrl);
+
+    if (!walletServiceWsUrl && _url.protocol.includes('https')) {
```

Comment:
I think it's better to require the user to set the ws URL as well, instead of assuming it would be this format. What do you think?

Reply:
Oh... We only infer the URL this way when they don't provides the WS URL, otherwise we use the URL from the input.

Reply:
Yes, I understand, but isn't it better to require the WS URL instead of inferring it?

Reply:
Making it required? Can be, for me, it's ok and cheap to introduce. My decision drive was to put the least possible barriers to not bother too much the dev.

Reply:
Changed at 562ee965ce5f0c68e0810fc16b81453893eece5c:
- explorerServiceUrl is required
- walletServiceWsUrl is conditionally required